src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -166,15 +165,15 @@ public void setupChild(Configuration childConf)
     }
 
     @Override
-    protected void processFiltered(File file, List<String> lines) throws CheckstyleException {
+    protected void processFiltered(File file, FileText fileText) throws CheckstyleException {
         // check if already checked and passed the file
         if (CommonUtils.matchesFileExtension(file, getFileExtensions())) {
             final String msg = "%s occurred during the analysis of file %s.";
             final String fileName = file.getPath();
             try {
                 if (!ordinaryChecks.isEmpty()
                         || !commentChecks.isEmpty()) {
-                    final FileText text = FileText.fromLines(file, lines);
+                    final FileText text = new FileText(fileText);
```

Comment:
@rnveach
There is no need in copying the file text instance, as I see it.

@timurt
Please, explain or remove the line.